src/ts/components/timelinemarkers.ts

```diff
@@ -0,0 +1,208 @@
+import { PlayerAPI } from 'bitmovin-player';
+import { UIInstanceManager } from '../uimanager';
+import { DOM } from '../dom';
+import { Component, ComponentConfig } from './component';
+import { TimelineMarker } from '../uiconfig';
+import { SeekBarMarker } from './seekbar';
+import { PlayerUtils } from '../playerutils';
+import { Timeout } from '../timeout';
+
+export interface MarkersConfig extends ComponentConfig {
+  /**
+   * Used for seekBar marker snapping range percentage
+   */
+  snappingRange?: number;
+}
+
+export class TimelineMarkers extends Component<MarkersConfig> {
+  private markersContainer: DOM;
+  private timelineMarkers: SeekBarMarker[];
+  private player: PlayerAPI;
+  private uimanager: UIInstanceManager;
+  private pausedTimeshiftUpdater: Timeout;
+  private getSeekBarWidth: () => number;
+  protected config: MarkersConfig;
+
+
+  constructor(config: MarkersConfig, getSeekBarWidth: () => number, markersContainer: DOM) {
+    super();
+    this.config = config;
+    this.getSeekBarWidth = getSeekBarWidth;
+    this.markersContainer = markersContainer;
+  }
+
+  public configure(player: PlayerAPI, uimanager: UIInstanceManager) {
+    this.player = player;
+    this.uimanager = uimanager;
+    this.configureMarkers();
+
+  }
+
+  private configureMarkers(): void {
+    // Remove markers when unloaded
+    this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, () => this.clearMarkers());
+    // Update markers when the size of the seekbar changes
+    this.player.on(this.player.exports.PlayerEvent.PlayerResized, () => this.updateMarkersDOM());
+
+    this.player.on(this.player.exports.PlayerEvent.SourceLoaded, () => {
+      if (this.player.isLive()) {
+        // Update marker position as timeshift range changes
+        this.player.on(this.player.exports.PlayerEvent.TimeChanged, () => this.updateMarkers());
+        // Update marker postion when paused as timeshift range changes
+        this.configureLivePausedTimeshiftUpdater(() => this.updateMarkers());
+      }
+    });
+    this.uimanager.getConfig().events.onUpdated.subscribe(() => this.updateMarkers());
+    this.uimanager.onRelease.subscribe(() => this.uimanager.getConfig().events.onUpdated.unsubscribe(() => this.updateMarkers()));
```

Comment:
The main reason why we use inline arrow functions everywhere is that we can pass them without a 'wrapper' arrow function to the `PlayerEventCallback`

Reply:
It gives you same result, I guess it depends on preference.

Reply:
👍